source/MRMesh/MRMeshBuilder.cpp

```diff
@@ -717,7 +717,7 @@ size_t duplicateNonManifoldVertices( Triangulation & t, FaceBitSet * region, std
     std::vector<IncidentVert> incidentItemsVector;
     preprocessTriangles( t, region, incidentItemsVector );
 
-    auto lastUsedVertId = incidentItemsVector.back().srcVert;
+    auto lastUsedVertId = lastValidVert.valid() ? lastValidVert : incidentItemsVector.back().srcVert;
```

Comment:
```suggestion
    if ( !lastValidVert )
        lastValidVert = incidentItemsVector.back().srcVert;
```
and use `lastValidVert` below